### PR TITLE
XHR FormData: Add tests for changes to the entry list during iteration.

### PR DESCRIPTION
**Pull request by rakuco**
This is similar to #36455 and #20445: we want to make sure that adding and deleting elements during iteration is reflected in FormData's value pairs to iterate over (i.e. iteration does not happen on a cached version).

### CODE REVIEW

File: xhr/formdata/iteration.any.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    const actualValues = [];
+    for (const [name, value] of formData) {
+        actualKeys.push(name);
+        actualValues.push(value);
+        if (name === "baz")
+            formData.delete("foo");
+    }
+    assert_array_equals(actualKeys, ["foo", "baz", "quux"]);
+    assert_array_equals(actualValues, ["0", "1", "3"]);
+}, "Removing elements already iterated over causes an element to be skipped during iteration");
```
[the remark this answers is not in the document]
Comment (rakuco) [reply xhr/formdata/iteration.any.js:45]: It's basically https://webidl.spec.whatwg.org/#es-iterator-prototype-object (specifically `next`'s steps, which don't seem to have <dfn> to link to.

I don't think there's a way to reference it in a .js test though.

[the remark this answers is not in the document]
Comment (Manishearth) [reply xhr/formdata/iteration.any.js:45]: oh just as a comment, i mean

[the remark this answers is not in the document]
Comment (rakuco) [reply xhr/formdata/iteration.any.js:48]: Sure thing. I've added a comment to the beginning of the file referencing that part of the Web IDL spec.

**Review by Manishearth — APPROVED** [review]
I meant to suggest more specific spec links in individual tests, but this is fine.

**Pull request merged**